lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/src/engine.dart';
+import 'package:ui/ui.dart' as ui;
+
+/// Tracks the [FlutterView]s focus changes.
+final class ViewFocusBinding {
+  ViewFocusBinding._();
+
+  static ViewFocusBinding? _instance;
+
+  /// Initializes the [ViewFocusBinding] singleton.
+  static void initialize() {
+    _instance ??= ViewFocusBinding._().._init();
+  }
```

Comment:
Dart lets you do prettier singletons, I think! :P

What about:

```dart
final class ViewFocusBinding {
  ViewFocusBinding({required EnginePlatformDispatcher dispatcher}) : _dispatcher = dispatcher {
    _init(); // Or move the code of init here...
  }

  static ViewFocusBinding instance = ViewFocusBinding();

  ...
```

The good thing about this code is that it lets you create as many ViewFocusBinding objects as you need for testing, and makes it very easy to override the singleton `instance` from tests, when you want to do integration tests of other stuff that might be using the `ViewFocusBinding.instance` directly.

(If there's no need to have an `instance` of this, and this doesn't need to be a singleton, just... remove the `instance` and keep a copy of the instance in the engine initialization so you can dispose of it later!

Reply:
Great call to simplify code overall.  You made me read https://dart.dev/language/class-modifiers which I really appreciate :handshake: 

I couldn't find any reference that suggested the `final` modifier could be used to prevent instantiation tho.